preprocess_reddit_dataset.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import csv
+import os
+import sys
+
+import tensorflow_datasets as tfds
+
+import preprocess_utils
+
+PREPROCESSED_FILE_PATH = "~/preprocessed_reddit_dataset.tsv"
+TRAIN_FILE_PATH = "~/train_reddit_dataset.tsv"
+TUNE_FILE_PATH = "~/tune_reddit_dataset.tsv"
+VALID_FILE_PATH = "~/valid_reddit_dataset.tsv"
+
+
+def main(argv):
+    """ Preprocess the Reddit dataset."""
+    if len(argv) != 2:
+        raise Exception("Usage: python preprocess_reddit_dataset num_of_tuning_samples num_of_validation_samples")
+
+    try:
+        num_of_tuning_sam = int(argv[0])
+        num_of_valid_sam = int(argv[1])
+    except ValueError:
+        raise Exception("Number of samples must be non-negative integers")
```

Comment:
you did not check non-negative. Also for other files.

Reply:
Done.